core/src/main/java/org/apache/iceberg/avro/BuildAvroProjection.java

```diff
@@ -96,7 +96,7 @@ public Schema record(Schema record, List<String> names, Iterable<Schema.Field> s
 
       } else {
         Preconditions.checkArgument(
-            field.isOptional() || field.fieldId() == MetadataColumns.ROW_POSITION.fieldId(),
+            field.isOptional() || MetadataColumns.metadataFieldIds().contains(field.fieldId()),
```

Comment:
Needed to allow projecting `_file` even though it isn't in the data file.

Reply:
This reminds me we need to fix that projection bug / selection bug